### DISCUSSION TITLE
Dynamic constraints possible?

### DISCUSSION THREAD

**Discussion by roangel** [facebookresearch/theseus#262] [Q&A] [answered]
Hello Theseus,

First, congrats for the amazing job! This library is very promising. I wanted to ask a simple question, I read the paper and it says that you don't support equality constraints in your optimization problem layers. Does this mean as well that users can't enforce dynamics (x_{k+1} = f(x_k, u_k)) in the optimization problem?

Thanks!

**Comment by mhmukadam** [comment, marked as answer]
Hi @roangel, thanks for your interest.

Yes you are correct, at the moment we only have support for unconstrained nonlinear least squares objectives. In practice typically we can just apply dynamics as a weighted soft constraint, || w (x_{k+1} - f(x_k, u_k)) ||^2 similar to [these cost functions](https://github.com/facebookresearch/theseus/blob/main/theseus/embodied/motionmodel/double_integrator.py) and are used in the motion planning example.

Support for constrained solvers is planned for the future.
